src/modules/rewards/services/opRebateServiceV2.ts

```diff
@@ -0,0 +1,347 @@
+        .execute();
+      const rewards = await entityManager
+        .createQueryBuilder()
+        .select("r")
+        .from(OpRewardV2, "r")
+        .where("r.windowIndex = :windowIndex", { windowIndex })
+        .getMany();
+      const { recipients, totalRewardsAmount } = this.aggregateOpRewards(rewards);
+
+      for (const recipientsChunk of splitArrayInChunks(recipients, 100)) {
+        await entityManager
+          .createQueryBuilder()
+          .insert()
+          .into(ReferralRewardsWindowJobResult)
+          .values(
+            recipientsChunk.map((recipient) => ({
+              jobId,
+              windowIndex,
+              totalRewardsAmount,
+              address: recipient.account,
+              amount: recipient.amount,
+            })),
+          )
+          .execute();
+      }
+    });
+  }
+
+  public aggregateOpRewards(rewards: OpRewardV2[]) {
+    // Map an address to considered deposits for referral rewards
+    const recipientOpRewardsMap = rewards.reduce((acc, r) => {
+      acc[r.recipient] = [...(acc[r.recipient] || []), r];
+      return acc;
+    }, {} as Record<string, OpRewardV2[]>);
+
+    let totalRewardsAmount: BigNumber = new BigNumber(0);
+    const recipients: {
+      account: string;
+      amount: string;
+    }[] = [];
+
+    for (const [address, opRewards] of Object.entries(recipientOpRewardsMap)) {
+      const rewardsAmount = opRewards.reduce((sum, opReward) => {
+        return sum.plus(opReward.amount);
+      }, new BigNumber(0));
+
+      totalRewardsAmount = totalRewardsAmount.plus(rewardsAmount);
+      recipients.push({
+        account: address,
+        amount: rewardsAmount.toFixed(),
+      });
+    }
+
+    return { totalRewardsAmount: totalRewardsAmount.toFixed(), recipients };
+  }
+
+  public isDepositEligibleForOpRewards(deposit: Pick<RewardedDeposit, "destinationChainId">) {
+    return ELIGIBLE_OP_REWARDS_CHAIN_IDS.includes(deposit.destinationChainId);
+  }
+
+  /**
+   * PRIVATE METHODS
+   */
+
+  private async calcOpRebateRewards(deposit: RewardedDeposit) {
+    // lp fee + relayer capital fee + relayer destination gas fee
+    const bridgeFeeUsd = new BigNumber(deposit.totalBridgeFeeUsd);
+    const rewardToken = await this.ethProvidersService.getCachedToken(
+      this.appConfig.values.rewardPrograms["op-rebates"].rewardToken.chainId,
+      this.appConfig.values.rewardPrograms["op-rebates"].rewardToken.address,
+    );
+
+    if (bridgeFeeUsd.lte(0)) {
+      return {
+        rewardToken,
+        rewardsUsd: "0",
+        rewardsAmount: new BigNumber(0),
+      };
+    }
+
+    const inputToken = await this.ethProvidersService.getCachedToken(deposit.originChainId, deposit.inputToken);
+    const outputToken = await this.ethProvidersService.getCachedToken(deposit.originChainId, deposit.outputToken);
+    
+    const inputTokenPrice = await this.getInputTokenPrice(deposit, inputToken, outputToken); // This could also be part of the indexer data
```

Comment:
This could also be part of the indexer data, but for now I'm fetching it here too.
This is needed to calculate the MAX reward amount for each deposit.